drivers/bluetooth/hci/h4.c

```diff
@@ -35,6 +35,17 @@
 #define H4_EVT  0x04
 #define H4_ISO  0x05
 
+
+/* Extern h4_vnd_setup function.
+ * This function executes vendor-specific commands sequence to
+ * initialize BT Controller before BT Host executes Reset sequence.
+ * h4_vnd_setup function must be implemented in vendor-specific HCI
+ * extansion module if CONFIG_BT_HCI_H4_SETUP is enabled.
+ */
+#ifdef CONFIG_BT_HCI_H4_SETUP
+extern int h4_vnd_setup(const struct device *dev);
+#endif
```

Comment:
The PR looks pretty good now, and I'm basically ready to approved it, however could you remind me why we wouldn't now want to go for a `__weak` function especially since `h4.c` already uses this tactic for another purpose. @carlescufi do you remember what was the reasoning for a Kconfig instead of `__weak`?

Reply:
Hi  @jhedberg, 
1. Details about of  propose to use "setup" is in https://github.com/zephyrproject-rtos/zephyr/issues/41140#issuecomment-996090075: 
```
We might want to consider replicating the Linux solution for this, where HCI drivers provide a setup() callback, which would get called some time after the open() callback
```
2. The discussion for implementation details is documented https://discord.com/channels/720317445772017664/733036956392357908/923585062170140715

Hi @carlescufi,
Could you please add your comments here.

Regards,
Nazar

Reply:
I am pretty sure the reason for this was to save the 4 (or 8) bytes for the `setup` function pointer in the `bt_hci_driver` struct. But right now this is not being conditionally compiled out, so it needs fixing.

Reply:
This would also be consistent with `conn.h`: https://github.com/zephyrproject-rtos/zephyr/blob/main/include/bluetooth/conn.h#L857

Reply:
@jhedberg does this make sense to you?